pkg/services/secrets/kvstore/fallback.go

```diff
@@ -0,0 +1,67 @@
+package kvstore
+
+import (
+	"context"
+
+	"github.com/grafana/grafana/pkg/infra/log"
+)
+
+type KVStoreWithFallback struct {
+	log      log.Logger
+	store    SecretsKVStore
+	fallback SecretsKVStore
+}
+
+func NewKVStoreWithFallback(store SecretsKVStore, fallback SecretsKVStore) *KVStoreWithFallback {
+	return &KVStoreWithFallback{
+		log:      log.New("secrets.kvstore"),
+		store:    store,
+		fallback: fallback,
+	}
+}
+
+func (kv *KVStoreWithFallback) Get(ctx context.Context, orgId int64, namespace string, typ string) (string, bool, error) {
+	value, ok, err := kv.store.Get(ctx, orgId, namespace, typ)
+	if err != nil || !ok {
+		value, ok, err := kv.fallback.Get(ctx, orgId, namespace, typ)
+		kv.log.Debug("failed to get secret with plugin, using fallback", "orgId", orgId, "type", typ, "namespace", namespace, "error", err)
+		return value, ok, err
+	}
+	return value, ok, err
+}
+
+func (kv *KVStoreWithFallback) Set(ctx context.Context, orgId int64, namespace string, typ string, value string) error {
```

Comment:
If this is meant to be a fallback when migration is occurring, I'm not sure it makes sense to actually implement set, del, and rename. For example, if you set a key, it won't be in the migrator's list of keys, so it won't get deleted and will stay forever. Therefore, perhaps all of the setters should just call the primary store.

Reply:
I also considered that.

Reply:
Done!

Reply:
I feel like `del` and `rename` would have different results if executed before/during/after migration. 
the problem with having the migration as a background service is that we can to be sure (for now) it is finished. 
How would `del` and `rename` will behave if the secrets are not yet migrated?